pinot-broker/src/main/java/org/apache/pinot/broker/routing/HelixExternalViewBasedTimeBoundaryService.java

```diff
@@ -48,77 +48,84 @@ public HelixExternalViewBasedTimeBoundaryService(ZkHelixPropertyStore<ZNRecord>
   }
 
   public void updateTimeBoundaryService(ExternalView externalView) {
-    if (_propertyStore == null) {
-      return;
-    }
-    String tableName = externalView.getResourceName();
-    // Do nothing for realtime table.
-    if (TableNameBuilder.getTableTypeFromTableName(tableName) == TableType.REALTIME) {
+    String tableNameWithType = externalView.getResourceName();
+
+    // Skip real-time table, only use offline table to update the time boundary
+    if (TableNameBuilder.getTableTypeFromTableName(tableNameWithType) == TableType.REALTIME) {
       return;
     }
 
     Set<String> offlineSegmentsServing = externalView.getPartitionSet();
     if (offlineSegmentsServing.isEmpty()) {
-      LOGGER.info("Skipping updating time boundary service for table '{}' with no offline segments.", tableName);
+      LOGGER.warn("Skip updating time boundary for table: '{}' with no offline segment", tableNameWithType);
       return;
     }
 
-    TableConfig offlineTableConfig = ZKMetadataProvider.getOfflineTableConfig(_propertyStore, tableName);
-    assert offlineTableConfig != null;
-    TimeUnit tableTimeUnit = offlineTableConfig.getValidationConfig().getTimeType();
-    if (tableTimeUnit == null) {
-      LOGGER.info("Skipping updating time boundary service for table '{}' because time unit is not set", tableName);
+    Schema schema = ZKMetadataProvider.getTableSchema(_propertyStore, tableNameWithType);
+    assert schema != null;
+    // TODO: support SDF
+    String timeColumn = schema.getTimeColumnName();
```

Comment:
I thought we were tending towards the time column definition in the tableConfig rather than schema (and all derived time columns in the schema). Please check with @npawar

Reply:
Discussed with @npawar. When we start using dateTime, pick the time column from table config, and use the DateTimeFieldSpec from the schema to determine the time unit. For now, added a TODO.